homeassistant/components/miele/button.py

```diff
@@ -0,0 +1,166 @@
+"""Platform for Miele button integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+import aiohttp
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN, PROCESS_ACTION, MieleActions, MieleAppliance
+from .coordinator import MieleConfigEntry, MieleDataUpdateCoordinator
+from .entity import MieleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MieleButtonDescription(ButtonEntityDescription):
+    """Class describing Miele button entities."""
+
+    press_data: dict[str, str | int | bool]
+
+
+@dataclass
+class MieleButtonDefinition:
+    """Class for defining button entities."""
+
+    types: tuple[MieleAppliance, ...]
+    description: MieleButtonDescription
+
+
+BUTTON_TYPES: Final[tuple[MieleButtonDefinition, ...]] = (
+    MieleButtonDefinition(
+        types=(
+            MieleAppliance.WASHING_MACHINE,
+            MieleAppliance.WASHING_MACHINE_SEMI_PROFESSIONAL,
+            MieleAppliance.TUMBLE_DRYER,
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.DISHWASHER,
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.WASHER_DRYER,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.STEAM_OVEN_MK2,
+            MieleAppliance.DIALOG_OVEN,
+        ),
+        description=MieleButtonDescription(
+            key="start",
+            translation_key="start",
+            press_data={PROCESS_ACTION: MieleActions.START},
```

Comment:
What's process action for? What more buttons will be added?

Reply:
"processAction" is a group of API commands. Start, Pause, and Stop are implemented as buttons. Start and stop Superfreeze mode on freezers is implemented as a switch. This is also how these controls are  implemented in the Miele apps. 

No more buttons are planned.

Reply:
Why don't we just pass on the action and have the PRESS_ACTION in the button implementation?